src/main/java/nerdhub/textilelib/mixins/MixinItemStack.java

```diff
@@ -37,4 +43,10 @@ public void useOnBlock(ItemUsageContext itemUsageContext_1, CallbackInfoReturnab
             }
         }
     }
+
+    @Inject(method = "getTooltipText", at = @At("RETURN"))
+    public void getTooltipText(@Nullable PlayerEntity playerEntity_1, TooltipOptions tooltipOptions_1, CallbackInfoReturnable cir) {
+        TooltipBuildEvent tooltipBuildEvent = new TooltipBuildEvent((ItemStack) (Object) this, (List<TextComponent>) cir.getReturnValue());
```

Comment:
Instead of casting the return value to List<TextComponent> you should rather add type parameters to the callback info